src/GitVersionExe.Tests/VersionWriterTests.cs

```diff
@@ -0,0 +1,56 @@
+﻿namespace GitVersionExe.Tests
+{
+    using System;
+    using System.IO;
+    using System.Reflection;
+    using GitVersion;
+    using Mono.Cecil;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class VersionWriterTests
+    {
+        [Test]
+        public void WriteVersion_ShouldWriteFileVersion_WithNoPrereleaseTag()
+        {
+            var asm = GenerateAssembly(new Version(1, 0, 0), "");
+
+            string version = null;
+            VersionWriter.WriteTo(asm, v => version = v);
+
+            Assert.IsNotNull(asm);
+            Assert.AreEqual("1.0.0", version);
+        }
+
+        [Test]
+        public void WriteVersion_ShouldWriteFileVersion_WithPrereleaseTag()
+        {
+            var asm = GenerateAssembly(new Version(1, 0, 0), "-beta0004");
+
+            string version = null;
+            VersionWriter.WriteTo(asm, v => version = v);
+
+            Assert.IsNotNull(asm);
+            Assert.AreEqual("1.0.0-beta0004", version);
+        }
+
+        private Assembly GenerateAssembly(Version fileVersion, string prereleaseInfo)
+        {
+            var definition = new AssemblyNameDefinition("test-asm", fileVersion);
+
+            var asmDef = AssemblyDefinition.CreateAssembly(definition, "test-asm", ModuleKind.Dll);
```

Comment:
It didn't actually, I didn't notice that before merging. I likely will just not run this test on Mono. Thoughts @asbjornu